services/venice-controller/src/main/java/com/linkedin/venice/controller/util/UpdateStoreUtils.java

```diff
@@ -0,0 +1,1173 @@
+    Optional<Boolean> readComputationEnabled = params.getReadComputationEnabled();
+    Optional<Integer> bootstrapToOnlineTimeoutInHours = params.getBootstrapToOnlineTimeoutInHours();
+    Optional<BackupStrategy> backupStrategy = params.getBackupStrategy();
+    Optional<Boolean> autoSchemaRegisterPushJobEnabled = params.getAutoSchemaRegisterPushJobEnabled();
+    Optional<Boolean> hybridStoreDiskQuotaEnabled = params.getHybridStoreDiskQuotaEnabled();
+    Optional<Boolean> regularVersionETLEnabled = params.getRegularVersionETLEnabled();
+    Optional<Boolean> futureVersionETLEnabled = params.getFutureVersionETLEnabled();
+    Optional<String> etledUserProxyAccount = params.getETLedProxyUserAccount();
+    Optional<Boolean> nativeReplicationEnabled = params.getNativeReplicationEnabled();
+    Optional<String> pushStreamSourceAddress = params.getPushStreamSourceAddress();
+    Optional<Long> backupVersionRetentionMs = params.getBackupVersionRetentionMs();
+    Optional<Integer> replicationFactor = params.getReplicationFactor();
+    Optional<Boolean> migrationDuplicateStore = params.getMigrationDuplicateStore();
+    Optional<String> nativeReplicationSourceFabric = params.getNativeReplicationSourceFabric();
+    Optional<Boolean> activeActiveReplicationEnabled = params.getActiveActiveReplicationEnabled();
+    Optional<String> personaName = params.getStoragePersona();
+    Optional<Map<String, String>> storeViewConfig = params.getStoreViews();
+    Optional<String> viewName = params.getViewName();
+    Optional<String> viewClassName = params.getViewClassName();
+    Optional<Map<String, String>> viewParams = params.getViewClassParams();
+    Optional<Boolean> removeView = params.getDisableStoreView();
+    Optional<Integer> latestSupersetSchemaId = params.getLatestSupersetSchemaId();
+    Optional<Boolean> storageNodeReadQuotaEnabled = params.getStorageNodeReadQuotaEnabled();
+    Optional<Long> minCompactionLagSeconds = params.getMinCompactionLagSeconds();
+    Optional<Long> maxCompactionLagSeconds = params.getMaxCompactionLagSeconds();
+    Optional<Integer> maxRecordSizeBytes = params.getMaxRecordSizeBytes();
+    Optional<Boolean> unusedSchemaDeletionEnabled = params.getUnusedSchemaDeletionEnabled();
+    Optional<Boolean> blobTransferEnabled = params.getBlobTransferEnabled();
+
+    owner.map(addToUpdatedConfigs(updatedConfigs, OWNER)).ifPresent(updatedStore::setOwner);
+    readability.map(addToUpdatedConfigs(updatedConfigs, ENABLE_READS)).ifPresent(updatedStore::setEnableReads);
+    writeability.map(addToUpdatedConfigs(updatedConfigs, ENABLE_WRITES)).ifPresent(updatedStore::setEnableWrites);
+    partitionCount.map(addToUpdatedConfigs(updatedConfigs, PARTITION_COUNT)).ifPresent(updatedStore::setPartitionCount);
+    largestUsedVersionNumber.map(addToUpdatedConfigs(updatedConfigs, LARGEST_USED_VERSION_NUMBER))
+        .ifPresent(updatedStore::setLargestUsedVersionNumber);
```

Comment:
Nitpick: This stuff looks a bit like boilerplate to me... I think we could simplify further, along the lines of:

```
  private static <T> void addToUpdatedConfigs(Set<CharSequence> updatedConfigs, String configName, Optional<T> param, Consumer<T> paramSetter) {
    if (param.isPresent()) {
      updatedConfigs.add(configName);
      paramSetter.accept(param.get());
    }
  }
```

And then:

```
  addToUpdatedConfigs(updatedConfigs, LARGEST_USED_VERSION_NUMBER, largestUsedVersionNumber, updatedStore::setLargestUsedVersionNumber);
```

It might still line wrap, depending on the length of the params we pass in, but the map/ifPresent ritual would be gone, which seems clearer to me...

Reply:
Let me try it. One more place that could benefit from some boilerplate removal is `VeniceParentHelixAdmin#updateStore`. Can you recommend some suggestions there as well?